cpp/src/arrow/compute/kernels/aggregate_basic.cc

```diff
@@ -339,13 +366,38 @@ std::unique_ptr<KernelState> MeanInit(KernelContext* ctx, const KernelInitArgs&
 template <typename ArrowType, typename Enable = void>
 struct MinMaxState {};
 
+template <typename ArrowType>
+struct MinMaxState<ArrowType, enable_if_boolean<ArrowType>> {
+  using ThisType = MinMaxState<ArrowType>;
+  using T = typename ArrowType::c_type;
+
+  ThisType& operator+=(const ThisType& rhs) {
+    this->has_nulls |= rhs.has_nulls;
+    this->has_values |= rhs.has_values;
```

Comment:
@wesm I had to introduce a flag to check whether any of the arrays have valid values, otherwise the default inverse extreme values were returned instead of null.